tests/aws/services/stepfunctions/v2/scenarios/test_base_scenarios.py

```diff
@@ -1546,6 +1546,36 @@ def test_retry_interval_features_jitter_none(
             exec_input,
         )
 
+    @markers.aws.validated
+    def test_retry_interval_features_max_attempts_zero(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        create_lambda_function,
+        sfn_snapshot,
+    ):
+        function_name = f"lambda_func_{short_uid()}"
+        sfn_snapshot.add_transformer(RegexTransformer(function_name, "lambda_function_name"))
+        create_lambda_function(
+            func_name=function_name,
+            handler_file=EHT.LAMBDA_FUNC_RAISE_EXCEPTION,
+            runtime="python3.12",
```

Comment:
nit: typed `Runtime.python3_12` would be nice (especially for maintenance updates at some point in the future)